src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -487,9 +487,7 @@ static bool IsAsyncMethod(MethodInfo method)
                 Throw.ArgumentException(nameof(parameter), "Parameter is missing a name.");
             }
 
-            // Resolve the contract used to marshal the value from JSON -- can throw if not supported or not found.
             Type parameterType = parameter.ParameterType;
-            JsonTypeInfo typeInfo = serializerOptions.GetTypeInfo(parameterType);
```

Comment:
Premature calculation of the JTI was causing errors because no contract could be resolved for `CancellationToken` or parameters using custom marshallers.